maker/src/lib.rs

```diff
@@ -14,6 +15,29 @@ pub mod cfd;
 mod metrics;
 pub mod routes;
 
+#[derive(Debug)]
+pub struct Password(String);
+
+impl From<[u8; 32]> for Password {
+    fn from(bytes: [u8; 32]) -> Self {
+        Self(hex::encode(bytes))
+    }
+}
+
+impl std::str::FromStr for Password {
+    type Err = Infallible;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(Self(s.to_owned()))
+    }
+}
+
+impl std::fmt::Display for Password {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
```

Comment:
no need for this duplication - you can easily move it to some shared location, e.g. `shared-bin/src/auth.rs`